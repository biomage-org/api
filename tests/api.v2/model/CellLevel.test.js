// @ts-nocheck
const { mockSqlClient } = require('../mocks/getMockSqlClient')();
const CellLevel = require('../../../src/api.v2/model/CellLevel');

jest.mock('../../../src/sql/sqlClient', () => ({
  get: jest.fn(() => mockSqlClient),
}));

describe('model/CellLevel', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('getMetadataByExperimentId retrieves data correctly', async () => {
    const mockExperimentId = 'mockExperimentId';
    const mockResult = {
      id: 'mockId',
      name: 'mockName',
      upload_status: 'mockStatus',
      created_at: 'mockDate',
    };

    mockSqlClient.first.mockReturnValueOnce(Promise.resolve(mockResult));

    const cellLevel = new CellLevel();
    const result = await cellLevel.getMetadataByExperimentId(mockExperimentId);

    expect(result).toEqual(mockResult);


    expect(mockSqlClient.select.mock.calls).toMatchSnapshot();
    expect(mockSqlClient.from.mock.calls).toMatchSnapshot();
    expect(mockSqlClient.leftJoin.mock.calls).toMatchSnapshot();
    expect(mockSqlClient.where.mock.calls).toMatchSnapshot();
  });
});
