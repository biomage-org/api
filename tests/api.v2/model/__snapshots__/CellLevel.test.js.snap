// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`model/CellLevel getMetadataByExperimentId retrieves data correctly 1`] = `
Array [
  Array [
    Array [
      "id",
      "name",
      "upload_status",
      "created_at",
    ],
  ],
]
`;

exports[`model/CellLevel getMetadataByExperimentId retrieves data correctly 2`] = `
Array [
  Array [
    "cell_metadata_file_to_experiment",
  ],
]
`;

exports[`model/CellLevel getMetadataByExperimentId retrieves data correctly 3`] = `
Array [
  Array [
    "cell_metadata_file",
    "cell_metadata_file_to_experiment.cell_metadata_file_id",
    "cell_metadata_file.id",
  ],
]
`;

exports[`model/CellLevel getMetadataByExperimentId retrieves data correctly 4`] = `
Array [
  Array [
    "cell_metadata_file_to_experiment.experiment_id",
    "mockExperimentId",
  ],
]
`;
